cogs/fun.py

```diff
@@ -119,15 +139,18 @@ async def on_raw_reaction_add(self, reaction):
         confession_channel = await self.bot.fetch_channel(self.config["CONFESSION"]["CHANNEL"])
         user_in_chan_guild = await confession_channel.guild.fetch_member(reaction.user_id)
 
-        # Ignore bots
-        if reaction_user.bot:
-            return
+        try:
+            last_confession_msg = await confession_channel.fetch_message(confession_channel.last_message_id)
+            new_count = int(last_confession_msg.content.split(":")[0]) + 1
+        except discord.errors.NotFound:
+            new_count = 1
 
-        # Check if "confirm mode" is enabled
-        if not self.confession_is_confirm_e:
+        if (reaction_user.bot  # Ignore bots
+            or self.confession_confirm_mode
+            or self.confession_msg.author.id != reaction_user.id):
             return
 
         # If reaction is in DM and user is present in confession channel, send msg
         if reaction_channel.type.name == "private" and user_in_chan_guild is not None:
-            self.confession_is_confirm_e = False
-            await confession_channel.send(self.confession_msg.content)
+            self.confession_confirm_mode = False
+            await confession_channel.send(f"{new_count}: {self.confession_msg.content}")
```

Comment:
Firstly, you should apply the lazy philosophy: make an early return to check if the reaction is on a correct message and in private, and only after fetch things to the discord API, or else you will just waste time and energy for the 99% unrelated reactions events that the bot receives.

Thereafter, the way you handle the "reaction queue" with your `confession_confirm_mode`and `confession_msg` is not really great. Following your code logic, only one person at a time can submit to a confession, or else it will break the system.
Consider using a dictionary to manage that, as I've done for the Github issue integration (#9).
I'd like to rework the code but I'm not familliar with the review system, so I'll maybe do a PR on your repo.